### PR TITLE
Remove `_deprecated_scope_block` on AssociationRelation

### PR DESCRIPTION
**Pull request by foxlance**
This fix addresses the `NoMethodError: undefined method _deprecated_scope_block for ActiveRecord::AssociationRelation` error one might encounter when using `.first_or_create`, and possibly `.first_or_initialize` after a `where` clause on an association, e.g.

```ruby
Facility.first.amenities.where(name: 'pool').first_or_create
NoMethodError: undefined method `_deprecated_scope_block' for #<ActiveRecord::AssociationRelation []>
```
This bug is specific to **Rails 6.1.x** with **protected_attributes_continued v1.7.0**. Downgrading to **protected_attributes_continued v1.6.0** works if one is not after any specific 1.7.0 features.

This fix is inspired by https://github.com/rails/rails/commit/255a7ef7b09ba96f92c7c8a8d59666314551f454.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,9 @@
 * Nothing yet
 * [View Full Diff](https://github.com/westonganger/protected_attributes_continued/compare/v1.7.0...master)
 
+## 1.8.0 - 2021-01-27
+
+* [#21](https://github.com/westonganger/protected_attributes_continued/pull/21) - Fix bug when calling `first_or_create` on an association relation.
```
Comment (westonganger) [inline CHANGELOG.md]: Could you just add this under the Unreleased section?

File: lib/protected_attributes/version.rb [hunk 2]
```diff
@@ -1,3 +1,3 @@
 module ProtectedAttributes
-  VERSION = "1.7.0".freeze
+  VERSION = "1.8.0".freeze
```
Comment (westonganger) [inline lib/protected_attributes/version.rb]: If you could revert this line that would be great. In the future Id recommend you do not change the version numbers in a PR.

File: lib/active_record/mass_assignment_security/association_relation.rb [hunk 3]
```diff
@@ -7,19 +7,16 @@ class AssociationRelation
       undef :create!
 
       def build(attributes = nil, options = {}, &block)
-        block = _deprecated_scope_block("new", &block)
-        scoping { @association.build(attributes, options, &block) }
+        @association.build(attributes, options, &block)
```
Comment (westonganger) [inline lib/active_record/mass_assignment_security/association_relation.rb]: Does this change behaviour for Rails 5.2 and below?

**Review by westonganger — COMMENTED** [review]
Thanks for this. I have left some comments.

File: lib/protected_attributes/version.rb (hunk 2 above)
Comment (foxlance) [reply lib/protected_attributes/version.rb]: Thanks! I'll keep that in mind.

File: CHANGELOG.md (hunk 1 above)
Comment (foxlance) [reply CHANGELOG.md]: Moved this under **Unreleased**. Let me know if I did it right.

File: lib/active_record/mass_assignment_security/association_relation.rb (hunk 3 above)
Comment (foxlance) [reply lib/active_record/mass_assignment_security/association_relation.rb]: It shouldn't as far as I can tell. There's already a test for this that can be found at `test/mass_assignment_security/association_relation_test.rb` and it passes the on 5.0, 5.1 and 5.2. But, just to be sure, I went back on our app that we've been upgrading from 5.1 all the way to 6.1, and using this version I can say that all our existing test passes.

**Comment by westonganger** [comment]
PR that makes the original change in behaviour - https://github.com/rails/rails/pull/29301 was merged on June 28, 2017. Was merged before Rails 5.1.3 but I think was only released for 5.2.0 (https://github.com/rails/rails/issues/33387).

PR which removes the scoping block - https://github.com/rails/rails/pull/33394 was merged on July 19, 2018. Would be included in Rails 5.2.1

I think that lets keep the `scoping` block for versions 5.1 and below. Can you add this change?

**Comment by foxlance** [comment]
@westonganger Done! Let me know if you find anything else worth addressing.

**Comment by westonganger** [comment]
@foxlance you added the scoping back for all Rails versions. We want the scoping block to be removed for Rails 5.2+.

**Comment by foxlance** [comment]
@westonganger Is the recent commit what you have in mind?

**Comment by westonganger** [comment]
Merged. Thanks for your contribution.

**Comment by westonganger** [comment]
v1.8.0 is now released which contains this fix.

**Pull request merged**